gstorm/include/detail/ranges/vector.h

```diff
@@ -319,28 +331,28 @@ struct gvector : public traits::range_forward_traits<T>, public range::gvector_b
   // host access
   // TODO: check if this is save or not
   typename T::value_type *data() {
-    auto accesor =
-        _buffer->template get_access<cl::sycl::access::mode::read_write, cl::sycl::access::target::host_buffer>();
-    return &accesor[0];
+    sycl_accessor_type accessor;
+    _cgh->require(*_buffer, accessor);
+    return &accessor[0];
   }
 
   const typename T::value_type *data() const {
-    auto accesor =
-        _buffer->template get_access<cl::sycl::access::mode::read_write, cl::sycl::access::target::host_buffer>();
-    return &accesor[0];
+    sycl_accessor_type accessor;
+    _cgh->require(*_buffer, accessor);
+    return &accessor[0];
   }
```

Comment:
These ought to be implemented in a common private static function.

Reply:
Why static?

Reply:
Not sure why you want that static @cjdb  ?

Reply:
Also, same problem as before, the value is not available after the scope ends